server/src/main/java/org/elasticsearch/indices/recovery/RecoverySourceHandler.java

```diff
@@ -364,10 +368,7 @@ public void phase1(final IndexCommit snapshot, final Supplier<Integer> translogO
                 cancellableThreads.execute(() ->
                         recoveryTarget.receiveFileInfo(response.phase1FileNames, response.phase1FileSizes, response.phase1ExistingFileNames,
                                 response.phase1ExistingFileSizes, translogOps.get()));
-                // How many bytes we've copied since we last called RateLimiter.pause
-                final Function<StoreFileMetaData, OutputStream> outputStreamFactories =
-                        md -> new BufferedOutputStream(new RecoveryOutputStream(md, translogOps), chunkSizeInBytes);
-                sendFiles(store, phase1Files.toArray(new StoreFileMetaData[phase1Files.size()]), outputStreamFactories);
+                cancellableThreads.executeIO(() -> sendFiles(store, phase1Files.toArray(new StoreFileMetaData[0]), translogOps));
```

Comment:
We should avoid accessing Lucene under cancellablethreads, see https://issues.apache.org/jira/browse/LUCENE-2239.

Just wrap the calls to `requestSeqIdTracker.waitForOpsToComplete` instead